classes/Hook/HookActionCarrierProcess.php

```diff
@@ -53,13 +53,14 @@ public function run()
             );
 
             $carrierName = $carrierRepository->findByCarrierId((int) $this->params['cart']->id_carrier);
-
-            if ((bool) Configuration::get('GA_V4_ENABLED')) {
-                $js = $this->getGoogleAnalytics4($carrierName);
-            } else {
-                $js = $this->getUniversalAnalytics($carrierName);
+            if (!empty($carrierName)) {
+                if ((bool) Configuration::get('GA_V4_ENABLED')) {
+                    $js = $this->getGoogleAnalytics4($carrierName);
+                } else {
+                    $js = $this->getUniversalAnalytics($carrierName);
+                }
+                $ganalyticsDataHandler->manageData($js, 'A');
```

Comment:
```suggestion
            if (empty($carrierName)) {
                return;
            }

            if ((bool) Configuration::get('GA_V4_ENABLED')) {
                $js = $this->getGoogleAnalytics4($carrierName);
            } else {
                $js = $this->getUniversalAnalytics($carrierName);
            }

            $ganalyticsDataHandler->manageData($js, 'A');
```

this way you have an early return and you don't have a nested `if` :) it improves code readability